subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/transport/RepositoryTransportFactory.java

```diff
@@ -108,7 +113,7 @@ public RepositoryTransport createTransport(Set<String> schemes, String name, Col
         ExternalResourceConnector resourceConnector = connectorFactory.createResourceConnector(connectionDetails);
         ExternalResourceCachePolicy cachePolicy = startParameterResolutionOverride.overrideExternalResourceCachePolicy(new DefaultExternalResourceCachePolicy());
 
-        return new ResourceConnectorRepositoryTransport(name, progressLoggerFactory, temporaryFileProvider, cachedExternalResourceIndex, timeProvider, cacheLockingManager, resourceConnector, buildOperationExecutor, cachePolicy);
+        return new ResourceConnectorRepositoryTransport(name, progressLoggerFactory, temporaryFileProvider, cachedExternalResourceIndex, timeProvider, cacheLockingManager, resourceConnector, buildOperationExecutor, cachePolicy, producerGuard);
```

Comment:
Don't tell me about this, I'm spending half of my time fixing tests for which the number of parameters in constructor changed :)